### ISSUE TITLE
Email bug in StartTLS and advanced settings

### ISSUE THREAD

**Issue by markkrj** [pwm-project/pwm#519]
SMTP has a bug introduced by 579c6ab

After trying out PWM 2.0 (master branch) with 1.9 config, SMTP started failing. Then I went on and investigated.

In 2.0 there was introduced two more options for SMTP configuration which are SMTPS (SMTP over SSL/TLS) and StartTLS (Plain SMTP with STARTTLS). But after trying everyone of them, I found out that the new StartTLS option does not work and plain SMTP does not use advanced settings anymore. We used StartTLS for over 3 years through advanced settings (mail.smtp.starttls.enable=true).

I went further and checked the code.
The following lines set SSL even if using StartTLS, which is wrong:
https://github.com/pwm-project/pwm/blob/57528e4edf806a45505abfdb79057f1aafdbc701/server/src/main/java/password/pwm/svc/email/EmailServerUtil.java#L175-L187

Also, the following is returning properties before putting advanced settings if using plain SMTP:
https://github.com/pwm-project/pwm/blob/820882b23d3ad8add0ecefc92bf35427c2d12a93/server/src/main/java/password/pwm/svc/email/EmailServerUtil.java#L171-L174

So basically, the only working options are plain SMTP without advanced settings and SMTPS.

I proposed the PR #494 for this in July and it's still hanging without a comment. It fix the problem and does not conflict, so why not merge and fix this once and for all?
